addons/base_address_extended/models/res_partner.py

```diff
@@ -17,123 +13,39 @@ class Partner(models.Model):
     street_number2 = fields.Char(
         'Door', compute='_compute_street_data', inverse='_inverse_street_data', store=True)
 
+    city_id = fields.Many2one('res.city', string='City ID')
+    country_enforce_cities = fields.Boolean(related='country_id.enforce_cities', readonly=True)
+
     def _inverse_street_data(self):
-        """Updates the street field.
-        Writes the `street` field on the partners when one of the sub-fields in STREET_FIELDS
-        has been touched"""
-        street_fields = self._get_street_fields()
+        """ update self.street based on street_name, street_number and street_number2 """
         for partner in self:
-            street_format = (partner.country_id.street_format or
-                '%(street_number)s/%(street_number2)s %(street_name)s')
-            previous_field = None
-            previous_pos = 0
-            street_value = ""
-            separator = ""
-            # iter on fields in street_format, detected as '%(<field_name>)s'
-            for re_match in re.finditer(r'%\(\w+\)s', street_format):
-                # [2:-2] is used to remove the extra chars '%(' and ')s'
-                field_name = re_match.group()[2:-2]
-                field_pos = re_match.start()
-                if field_name not in street_fields:
-                    raise UserError(_("Unrecognized field %s in street format.", field_name))
-                if not previous_field:
-                    # first iteration: add heading chars in street_format
-                    if partner[field_name]:
-                        street_value += street_format[0:field_pos] + partner[field_name]
-                else:
-                    # get the substring between 2 fields, to be used as separator
-                    separator = street_format[previous_pos:field_pos]
-                    if street_value and partner[field_name]:
-                        street_value += separator
-                    if partner[field_name]:
-                        street_value += partner[field_name]
-                previous_field = field_name
-                previous_pos = re_match.end()
-
-            # add trailing chars in street_format
-            street_value += street_format[previous_pos:]
-            partner.street = street_value
+            street = ((partner.street_name or '') + " " + (partner.street_number or '')).strip()
+            if partner.street_number2:
+                street = street + " - " + partner.street_number2
+            partner.street = street
```

Comment:
In my opinion, this is making this module completely useless... There is different street formats all around the world. We can't just drop it like this. Do we plan to add manually all street formatting later like
```py
if partner.country_code == 'IN':
    street_format = ...
elif ...
```
?